sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs

```diff
@@ -1054,5 +1056,291 @@ public virtual Response<SearchSuggestion[]> GetSearchSuggestions(string searchSt
+        /// <param name="options">Optional parameters to use when querying for events.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The pageable list <see cref="AsyncPageable{QueryResultPage}"/> of query result frames.</returns>
+        public virtual AsyncPageable<QueryResultPage> QueryEventsAsync(
+            TimeSeriesId timeSeriesId,
+            DateTimeOffset startTime,
+            DateTimeOffset endTime,
+            QueryEventsRequestOptions options = null,
+            CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(QueryEvents)}");
+            scope.Start();
+
+            try
+            {
+                return QueryEventsInternalAsync(timeSeriesId, startTime, endTime, options, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Retrieve raw events for a given Time Series Id synchronously.
+        /// </summary>
+        /// <param name="timeSeriesId">The Time Series Id to retrieve raw events for.</param>
+        /// <param name="startTime">Start timestamp of the time range. Events that have this timestamp are included.</param>
+        /// <param name="endTime">End timestamp of the time range. Events that match this timestamp are excluded.</param>
+        /// <param name="options">Optional parameters to use when querying for events.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The pageable list <see cref="AsyncPageable{QueryResultPage}"/> of query result frames.</returns>
+        public virtual Pageable<QueryResultPage> QueryEvents(
+            TimeSeriesId timeSeriesId,
+            DateTimeOffset startTime,
+            DateTimeOffset endTime,
+            QueryEventsRequestOptions options = null,
+            CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(QueryEvents)}");
+            scope.Start();
+
+            try
+            {
+                return QueryEventsInternal(timeSeriesId, startTime, endTime, options, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Retrieve raw events for a given Time Series Id over a specified timespan asynchronously.
+        /// </summary>
+        /// <param name="timeSeriesId">The Time Series Id to retrieve raw events for.</param>
+        /// <param name="timeSpan">The timespan over which to query data.</param>
```

Comment:
not strong opinion but I would like to reword to "... time **interval** over which to query data.